### DISCUSSION TITLE
Option to index/categorize recipies by book

### DISCUSSION THREAD

**Discussion by DYLaKo** [mealie-recipes/mealie#712] [Feature Request]
**Before submitting this feature request I have**
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
I have a lot of recipe books and I miss an overview of all the recipies in the books. For example: I know I have multiple recipies with chicken, but I dont know which book contains them. If there was an index, I could enter the search keyword "chicken" and get presented with all the recipies containing chicken, which book it contains and what page. 

**(Optional): Suggest A Solution**
This could be the same as the categories and labels, but a new option called "Source" or something like that. Once a book is added, it will be available in the dropdown list at this new option and you can filter on the "source".
Next to that, it would be nice if you can enter the pagenumber where the recipe can be found in the book.

**Additional Information**
There are some online services which do such thing, but require a subscription and miss out the option to add complete recipies, they only do index:
International : https://www.eatyourbooks.com/
Dutch : https://heerlijkzoeken.nl/
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by hay-kot** [comment]
I'm not sure I understand what you're asking. Are you wanting a completely separate section on the site for indexing books? As in a "Cookbook Index" page where you can input just the title to recipes, their books, and page numbers, and then tag them as you wish? 

To me, it seems like this can pretty much already be accomplished by using recipes and adding the references to the books in the description and tagging accordingly.

**Reply by DYLaKo** [reply]
> I'm not sure I understand what you're asking. Are you wanting a completely separate section on the site for indexing books? As in a "Cookbook Index" page where you can input just the title to recipes, their books, and page numbers, and then tag them as you wish?

It doesn't need to be a completely separate section, but some fields added to the current recipies to quickly identify which book and which page it can be found would make this possible.

> To me, it seems like this can pretty much already be accomplished by using recipes and adding the references to the books in the description and tagging accordingly.

True, but one of the disadvantages of using the description for this is that there is a higher chance of inconsistency throughout all the recipies. With a fixed field to enter a pagenumber and a dropdown like solution for the book (like categories is now), this chance is lower.

**Reply by hay-kot** [reply]
I don't think the specific use-case warrants the additional work required to make this a full fledged feature. If this requests get a lot of traction with other users I may reconsider. 

Thanks!

**Comment by epheterson** [comment]
Actually thought about this also, would be nice to be able to group recipes by source. We have some recipe websites that have recipes of a certain theme, or some we'll be copying out of books, and it'd be great to track all recipes from one source. I'd use the page number info less, but a good idea as well. I was thinking the same, a third type of tag where you can put a source in.

**Comment by epheterson** [comment]
If this does get picked up, it'd be cool if the source could be associated with a URL, and Mealie could automatically scan existing recipes/assign on new recipes for URLs that are related.
